src/pg/sql/41_observatory_augmentation.sql

```diff
@@ -0,0 +1,995 @@
+
+--For Longer term Dev
+
+
+--Break out table definitions to types
+--Automate type creation from a script, something like
+----CREATE OR REPLACE FUNCTION OBS_Get<%=tag_name%>(geom GEOMETRY)
+----RETURNS TABLE(
+----<%=get_dimensions_for_tag(tag_name)%>
+----AS $$
+----DECLARE
+----target_cols text[];
+----names text[];
+----vals numeric[];-
+----q text;
+----BEGIN
+----target_cols := Array[<%=get_dimensions_for_tag(tag_name)%>],
+
+
+--Functions for augmenting specific tables
+--------------------------------------------------------------------------------
+
+-- Creates a table of demographic snapshot
+-- TODO: Remove since it does address geocoding?
+CREATE OR REPLACE FUNCTION OBS_GetDemographicSnapshot(address text)
+RETURNS TABLE(
+  total_pop NUMERIC,
+  male_pop NUMERIC,
+  female_pop NUMERIC,
+  median_age NUMERIC,
+  white_pop NUMERIC,
+  black_pop NUMERIC,
+  asian_pop NUMERIC,
+  hispanic_pop NUMERIC,
+  amerindian_pop NUMERIC,
+  other_race_pop NUMERIC,
```

Comment:
We are working on a way of generating these automatically using templates whenever the metadata updates. Only way we can figure out to get a table returning function as there is no way of having a table returning function with a dynamic return signature.